hw/drivers/bq27z561/src/bq27z561.c

```diff
@@ -440,12 +490,181 @@ bq27z561_get_temp(struct bq27z561 *dev, float *temp_c)
+            (uint8_t *)temp_c);
+
+    return rc;
+}
+
+int
+bq27z561_set_temp_lo_clr_threshold(struct bq27z561 *dev, int8_t temp_c)
+{
+    int rc;
+
+    rc = bq27z561_wr_std_reg_byte(dev, BQ27Z561_REG_TEMP_LO_CLR_TH,
+            (uint8_t)temp_c);
+
+    return rc;
+}
+
+int
+bq27z561_get_temp_hi_set_threshold(struct bq27z561 *dev, int8_t *temp_c)
+{
+    int rc;
+
+    rc = bq27z561_rd_std_reg_byte(dev, BQ27Z561_REG_TEMP_HI_SET_TH,
+            (uint8_t *)temp_c);
+
+    return rc;
+}
+
+int
+bq27z561_set_temp_hi_set_threshold(struct bq27z561 *dev, int8_t temp_c)
+{
+    int rc;
+
+    rc = bq27z561_wr_std_reg_byte(dev, BQ27Z561_REG_TEMP_HI_SET_TH,
+            (uint8_t)temp_c);
+
+    return rc;
+}
+
+int
+bq27z561_get_temp_hi_clr_threshold(struct bq27z561 *dev, int8_t *temp_c)
+{
+    int rc;
+
+    rc = bq27z561_rd_std_reg_byte(dev, BQ27Z561_REG_TEMP_HI_CLR_TH,
+            (uint8_t *)temp_c);
+
+    return rc;
+}
+
+int
+bq27z561_set_temp_hi_clr_threshold(struct bq27z561 *dev, int8_t temp_c)
+{
+    int rc;
+
+    rc = bq27z561_wr_std_reg_byte(dev, BQ27Z561_REG_TEMP_HI_CLR_TH,
+            (uint8_t)temp_c);
+
+    return rc;
+}
+
 int
 bq27z561_get_voltage(struct bq27z561 *dev, uint16_t *voltage)
 {
     int rc;
 
     rc = bq27z561_rd_std_reg_word(dev, BQ27Z561_REG_VOLT, voltage);
+
+    return rc;
+}
+
+int
+bq27z561_get_voltage_lo_set_threshold(struct bq27z561 *dev, uint16_t *voltage)
+{
+    int rc;
+
+    rc = bq27z561_rd_std_reg_word(dev, BQ27Z561_REG_VOLT_LO_SET_TH, voltage);
+
+    return rc;
+}
+
+int
+bq27z561_set_voltage_lo_set_threshold(struct bq27z561 *dev, uint16_t voltage)
+{
+    int rc;
+
+    rc = bq27z561_wr_std_reg_word(dev, BQ27Z561_REG_VOLT_LO_SET_TH, voltage);
+
+    return rc;
+}
+
+int
+bq27z561_get_voltage_lo_clr_threshold(struct bq27z561 *dev, uint16_t *voltage)
+{
+    int rc;
+
+    rc = bq27z561_rd_std_reg_word(dev, BQ27Z561_REG_VOLT_LO_CLR_TH, voltage);
+
+    return rc;
+}
+
+int
+bq27z561_set_voltage_lo_clr_threshold(struct bq27z561 *dev, uint16_t voltage)
+{
+    int rc;
+
+    rc = bq27z561_wr_std_reg_word(dev, BQ27Z561_REG_VOLT_LO_CLR_TH, voltage);
+
+    return rc;
+}
+
+int
+bq27z561_get_voltage_hi_set_threshold(struct bq27z561 *dev, uint16_t *voltage)
+{
+    int rc;
+
+    rc = bq27z561_rd_std_reg_word(dev, BQ27Z561_REG_VOLT_HI_SET_TH, voltage);
+
+    return rc;
+}
+
+int
+bq27z561_set_voltage_hi_set_threshold(struct bq27z561 *dev, uint16_t voltage)
+{
+    int rc;
+
+    rc = bq27z561_rd_std_reg_word(dev, BQ27Z561_REG_VOLT_HI_SET_TH, &voltage);
+
+    return rc;
+}
+
+int
+bq27z561_get_voltage_hi_clr_threshold(struct bq27z561 *dev, uint16_t *voltage)
+{
+    int rc;
+
+    rc = bq27z561_rd_std_reg_word(dev, BQ27Z561_REG_VOLT_HI_CLR_TH, voltage);
+
+    return rc;
+}
+
+int
+bq27z561_set_voltage_hi_clr_threshold(struct bq27z561 *dev, uint16_t voltage)
+{
+    int rc;
+
+    rc = bq27z561_wr_std_reg_word(dev, BQ27Z561_REG_VOLT_HI_CLR_TH, voltage);
+
     return rc;
```

Comment:
I am not saying that this is right or wrong, but I wonder what the compiler is going to do with the above functions. I do not trust compilers so part of the reason that these "helper wrappers" were not added in the original driver is that they add a bit of code just to hide the underlying function calls. All of these could be made macros as well. The use of static inline (and even always inline) does not appear to be a guarantee these things are inlined. Not a big deal but still.